src/Algolia.Search/Transport/HttpTransport.cs

```diff
@@ -134,7 +137,7 @@ public HttpTransport(AlgoliaConfig config, IHttpRequester httpClient)
                 }
             }
 
-            throw new AlgoliaUnreachableHostException("Unreachable hosts");
+            throw new AlgoliaUnreachableHostException(error);
```

Comment:
I would write a more user-friendly message like: `RetryStrategy failed to connect to Algolia. Reason: [...]`

Reply:
@TomKlotzPro  It could be a topic you'd like to discuss within the team in order to have a common message across the different API Clients implem.

Reply:
Totally agree, good idea for developper to have a better stacktrace. This result to confusion to just have "Unreachable hosts". Recently faced this for ruby and python support.